api/server/httputils/errors.go

```diff
@@ -72,6 +73,12 @@ func GetHTTPErrorStatusCode(err error) int {
 				break
 			}
 		}
+
+		// if the error has a type of grpc.rpcError, the error message should
+		// contain a sub string of "rpc error: code = "
+		if strings.Contains(errStr, "rpc error: code = ") {
```

Comment:
Can we move this up to top of the default case and instead pass the error directly rather than checking the error string?

Then we can gate the horrid string checking pattern currently above this with something like:
```go
if statusCode == http.StatusInternalServerError {
    // horrible string parser
}
```